### PR TITLE
Add `default_blob_access_tier` operator to storage builder

### LINKED ISSUE #454: Storage account access tier 'cool'
Is there any way to set the access tier on a storage account object? My understanding is that there is a significant cost difference between coold and hot tier access so would like to configure the tier to reduce cost.

```fsharp
  let storageAcc = storageAccount {
    name "myStorageAccount" 
    sku Storage.Sku.Standard_LRS
    
    use_static_website contentPath "index.html"
  }
```
produces
```json
    {
      "apiVersion": "2019-06-01",
      "dependsOn": [],
      "kind": "StorageV2",
      "location": "uksouth",
      "name": "myStorageAccount",
      "properties": {},
      "sku": {
        "name": "Standard_LRS"
      },
      "type": "Microsoft.Storage/storageAccounts"
    }
```
I believe the difference I would like is
```diff
      "location": "uksouth",
      "name": "myStorageAccount",
-     "properties": {},
+     "properties": {
+        "accessTier": "Cool"
+     },
      "type": "Microsoft.Storage/storageAccounts"
    }
```

### PR DESCRIPTION
**Pull request by TheRSP**
This PR closes #454

The changes in this PR are as follows:

* Added `default_blob_access_tier` operator to storage builder

I have read the [contributing guidelines](CONTRIBUTING.md) and have completed the following:

* [x] **Tested my code** end-to-end against a live Azure subscription.
* [x] **Updated the documentation** in the docs folder for the affected changes.
* [x] **Written unit tests** against the modified code that I have made.
* [x] **Updated the [release notes](RELEASE_NOTES.md)** with a new entry for this PR.
* [x] **Checked the coding standards** outlined in the [contributions guide](CONTRIBUTING.md) and ensured my code adheres to them.

If I haven't completed any of the tasks above, I include the reasons why here:

### CODE REVIEW

File: src/Tests/Storage.fs [hunk 1]
```diff
@@ -183,4 +183,21 @@ let tests = testList "Storage Tests" [
...
         Expect.equal resource.Sku.Name "Premium_LRS" "Sku Name"
     }
+    test "Sets blob access tier correctly different SKU kinds correctly" {
+        let account = storageAccount { default_blob_access_tier Cool }
+        let resource = arm { add_resource account } |> getStorageResource
+        Expect.equal resource.AccessTier (Nullable AccessTier.Cool) "Access Tier"
+
+        let account = storageAccount { default_blob_access_tier Hot }
+        let resource = arm { add_resource account } |> getStorageResource
+        Expect.equal resource.AccessTier (Nullable AccessTier.Cool) "Access Tier"
```
Comment (isaacabraham) [inline src/Tests/Storage.fs:193]: Shouldn't this be Hot?

File: src/Farmer/Arm/Storage.fs [hunk 2]
```diff
@@ -16,6 +16,7 @@ type StorageAccount =
     { Name : StorageAccountName
       Location : Location
       Sku : Sku
+      DefaultBlobAccessTier: BlobAccessTier Option
```
Comment (isaacabraham) [inline src/Farmer/Arm/Storage.fs:19]: I'm not sure at the ARM level that we should be moving this (probably makes sense at the Builder level though). Does that default access tier apply across all different storage kinds e.g. General Purpose, Blobs, Files etc. or just a subset of them?

File: src/Tests/Storage.fs (hunk 1 above)
Comment (TheRSP) [reply src/Tests/Storage.fs:193]: Yes. Don't know how that slipped through :(

File: src/Farmer/Arm/Storage.fs (hunk 2 above)
Comment (TheRSP) [reply src/Farmer/Arm/Storage.fs:19]: Looking into this further, while it's valid to set for all kinds except StorageV1, it only seems to affect StorageV2 and Blob Storage. 
Would you prefer adding the accessTier to the storageV2 sku?
Given that the SKU is used at both arm and builder levels, I'm not sure how best to separate those?

Comment (isaacabraham) [reply src/Farmer/Arm/Storage.fs:19]: Mmmm. Either we'll need a new type just for the builder (seems overkill) or we put that default flag against just those two kinds at both builder + ARM levels. We can still expose it as a keyword at the builder level - if someone tries to set it against an invalid storage account kind, throw an exception.

**Comment by isaacabraham** [comment]
Hi @TheRSP. I've pushed to the PR branch per the discussion above. Can you have a look through if you're happy with this? The only real changes are moving the default access tier into those two storage kinds, and also renaming the DU (there are three access tiers, but only two for default settings).

**Comment by TheRSP** [comment]
@isaacabraham This all looks good to me. Thanks.

**Comment by isaacabraham** [comment]
I'll merge this in once 1.3.x is released.

**Comment by isaacabraham** [comment]
Thanks again for doing this.

**Pull request merged**